sdks/python/apache_beam/pipeline.py

```diff
@@ -525,6 +525,31 @@ def run(self, test_runner_api='AUTO'):
     self.contains_external_transforms = (
         ExternalTransformFinder.contains_external_transforms(self))
 
+    # Finds if RunInference has side inputs enables.
```

Comment:
We should not be adding special support in pipeline.py for specific transforms. (E.g. this code will not get run when called via a cross-language transform.) Instead, we should be doing these checks at pipeline construction time (which will result in better, more local errors for the user too).

Reply:
I moved away from the `pipeline.py` and added the logic during pipeline construction time. Thanks for the tip.